scaffolding-chef/lib/scaffolding.sh

```diff
@@ -52,8 +52,11 @@ export SSL_CERT_FILE="{{pkgPathFor "core/cacerts"}}/ssl/cert.pem"
 cd {{pkg.path}}
 
 exec 2>&1
+while true; do
 chef-client -z -l {{cfg.log_level}} -c $pkg_svc_config_path/client-config.rb --once
-exec chef-client -z -i {{cfg.interval}} -s {{cfg.splay}} -l {{cfg.log_level}} -c $pkg_svc_config_path/client-config.rb
+sleep {{cfg.interval}}
+sleep {{cfg.splay}}
```

Comment:
@echohack this review isn't blocking this PR, just something for your consideration.
If lines 56-58 were the following:

```
sleep {{cfg.splay}}
chef-client -z -l {{cfg.log_level}} -c $pkg_svc_config_path/client-config.rb --once --no-fork --run-lock-timeout {{cfg.interval}}
sleep {{cfg.interval}}
```

It would give you these benefits:
 • `--no-fork` will help prevent zombie processes over the life of the service. Without it, the client will fork a subprocess. During the life of that child process, if the launcher kills the Supervisor manager for the chef-client, it can leave the child process without a parent, aka a zombie pid.
• `--run-lock-timeout {{cfg.interval}}` will give you an additional safety net to prevent a scenario where the Supervisor manager is terminated while a converge is running; leaving a stale lock file behind and preventing the next client from ever starting up. The help option describes it as such: "Set maximum duration to wait for another client run to finish, default is indefinitely."
• Separating the sleeps with the line: `sleep {{cfg.splay}}` first, before the client is run, will help mitigate thundering herd situations when large environments are provisioned. Otherwise without this, the operator will need to account for staggering bootstraps or batch provisioning of large environments which could be out of their ability to control. I understand that one desired outcome (probably mostly during demos) is for the client to converge immediately. This can should be easily accomplished during provisioning with a `user.toml` file containing `splay = 0`.

Reply:
Hell yeah, I will pull in all of these changes. THANK YOU!

Reply:
Just a note here for the future. It looks like --no-fork and --once will end up triggering the same `if` statement in the chef-client ruby code, so we were already safe from this one. However setting both will make us future-proof.